### ISSUE TITLE
No file formats?

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-aggregator#1]
Right now /file_formats doesn't expose any file formats. Is that intentional?

**Comment by jdries** [comment]
No, can be fixed. I propose to return the union of file formats and their parameters. (And evaluate how that looks.)
Backends will then throw an error if a certain format or parameter is not supported, which will point to further alignment needs.

**Comment by soxofaan** [comment]
Just pushed an initial/naive implementation (union of formats)

Two questions that already came up:
- VITO backend defines output format `NetCDF`, while EODC defines `netCDF`. Current implementation of aggregator is case sensitive and will list both `NetCDF` and `netCDF`, which is not ideal of course. 
  - openEO spec says "Format names and parameters MUST be fully aligned with the GDAL codes if available", and "NetCDF" is listed there as `netCDF` so I guess the quick fix is to change to `netCDF` in VITO backend. 
  - Still, it would be better to enforce that format names are to be handled in case insensitive manner. API at the moment just notes "Format names are allowed to be case insensitive throughout the API." which is not very enforcing
- for `GTiff` format VITO lists a couple of (custom) parameters (ZLEVEL, tile_grid, feature_id_property, sample_by_feature), while EODC has none. I'm not sure at the moment what the best approach of merging this is.

**Comment by soxofaan** [comment]
(internal ticket EP-3984)

**Comment by m-mohr** [comment]
I can confirm that this works on the homepage: https://docs.openeo.cloud/file-formats/

I agree that VITO should expose NetCDF as `netCDF`. If the VITO implementation is API compliant, this shouldn't break anything as format names must be accepted case-insensitive anyway in processes etc.

> Still, it would be better to enforce that format names are to be handled in case insensitive manner. API at the moment just notes "Format names are allowed to be case insensitive throughout the API." which is not very enforcing

What do you mean?

> for GTiff format VITO lists a couple of (custom) parameters (ZLEVEL, tile_grid, feature_id_property, sample_by_feature), while EODC has none. I'm not sure at the moment what the best approach of merging this is.

Long-term these should be aligned between backends and for those which are only present on one back-end maybe only list what is available on both? For now, maybe just add something to the description that states "Parameter only available for collections provided by VITO and Sinergise"?

**Comment by soxofaan** [comment]
> >  API at the moment just notes "Format names are allowed to be case insensitive throughout the API." which is not very enforcing
>
> What do you mean?

sorry I was looking at the master version of the spec, we already addressed my concern  in https://github.com/Open-EO/openeo-api/issues/371 : 

>         Format names MUST be accepted in a *case insensitive* manner throughout the API.

**Comment by soxofaan** [comment]
The `netCDF/NetCDF` issue should now be addressed.

**Comment by soxofaan** [comment]
basic implementation is in place now

Still open or up for discussion, but lower priority
- currently the union of file formats across backends is returned, for pragmatic reasons. However in principle the intersection should be taken
- regarding format parameters: current implementation takes parameters from first/primary backend (VITO at the moment). In principle the intersection of parameters should be taken
